bioptim/dynamics/configure_problem.py

```diff
@@ -312,14 +348,75 @@ def configure_contact_function(ocp, nlp, dyn_func: Callable, **extra_params):
             legend=all_contact_names,
         )
 
+    @staticmethod
+    def configure_soft_contact_function(ocp, nlp):
+        """
+        Configure the soft contact sphere
+
+        Parameters
+        ----------
+        ocp: OptimalControlProgram
+            A reference to the ocp
+        nlp: NonLinearProgram
+            A reference to the phase
+        """
+
+        global_soft_contact_force_func = MX.zeros(nlp.model.nbSoftContacts() * 6, 1)
+
+        for i_sc in range(nlp.model.nbSoftContacts()):
+            soft_contact = nlp.model.softContact(i_sc)
+            n = int(nlp.states.mx_reduced.shape[0] / 2)
```

Comment:
this seems very sketchy? What is that for? I feel that you make the assumption that states is q/qdot and you therefore get nq, if so, please just do nlp.model.nbQ()
In any case, there is no way you can trust the size of states